flaml/automl.py

```diff
@@ -1106,10 +1160,17 @@ def custom_metric(
         if self._state.task == 'classification':
             self._state.task = get_classification_objective(
                 len(np.unique(self._y_train_all)))
-            assert split_type in ["stratified", "uniform"]
+            assert split_type in ["stratified", "uniform", "time"]
             self._split_type = split_type
-        else:
-            self._split_type = "uniform"
+        elif self._state.task == 'regression':
+            if split_type in ["uniform", "time"]:
+                self._split_type = split_type
+            else:
+                self._split_type = "uniform"
+        elif self._state.task == 'forecast':
+            self._split_type = "time"

```

Comment:
`split_type` is a function's argument but it is silently overridden here. If the split type is always "time" when the task is "forecast", suggest raising an exception when the split_type is set (not None) but not to "time", and explain the reason in the error message.

Reply:
got it, changes made locally